### PR TITLE
Add DG part 2, 3 and 4

### PR DESCRIPTION
**Pull request by leezhixuan**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -285,14 +285,48 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+    
+    * 2a1. Connections show an error message.
+      
+      Use case resumes at step 2
+    
+* 2b. User provides details identical to an existing entry in Connections
+
+    * 2b1. Connections remind User that this is a duplicate
+    
+      Use case resumes at Step 1
```
Comment (yucheng11122017) [inline docs/DeveloperGuide.md:310]: Should this be at Step 2 instead?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -285,14 +285,48 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+    
+
+**Use case: Clear all entries**
+
+**MSS**
+
+1.  User request to clear all entries
+2.  Connections becomes empty
```
Comment (yucheng11122017) [inline docs/DeveloperGuide.md:318]: Maybe a better way to phrase this is "Connections deletes all the contacts"?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -285,14 +285,48 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+    Use case ends.
+
 **Use case: Delete a person**
 
 **MSS**
 
 1.  User requests to list persons
-2.  AddressBook shows a list of persons
+2.  Connections show a list of persons
```
Comment (yucheng11122017) [inline docs/DeveloperGuide.md:327]: Isn't it shows since it is singular?

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -308,6 +342,180 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
       Use case resumes at step 2.
 
+**Use case: Edit a person's details**
+
+**MSS**
+
+1.  User requests to list persons
+2.  Connections show a list of persons
+3.  User provides his index in the list along with the replacement information
```
Comment (yucheng11122017) [inline docs/DeveloperGuide.md:351]: Maybe "User provides the index of person to edit.." is better?

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -308,6 +342,180 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+2.  Use provides the search term
+3.  Connections return all entries that matches the search term
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. Connections is empty
+
+    * 2a1. Connections display a message to indicate no matching entries
```
Comment (yucheng11122017) [inline docs/DeveloperGuide.md:393]: Maybe display there are no entries is better to differentiate from here and the case below?

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -308,6 +342,180 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+2.  User provides the tag
+3.  Connections return all entries that matches the tag
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. Connections is empty
+
+    * 2a1. Connections display a message to indicate no matching entries
```
Comment (yucheng11122017) [inline docs/DeveloperGuide.md:417]: Same as comment above

File: docs/team/leezhixuan.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
+---
+layout: page
+title: Lee Zhi Xuan's Project Portfolio Page
```
Comment (cookiedan42) [inline docs/team/leezhixuan.md:3]: edit entry before merging

**Review by cookiedan42 — APPROVED** [review]

**Review by yucheng11122017 — APPROVED** [review]

**Pull request merged**